tests/dockerConfiguration/publicAddressVariable_test.js

```diff
@@ -0,0 +1,109 @@
+  await I.wait(30);
+};
+
+const runContainerWithPublicAddressVariableUpgrade = async (I, publicAddress) => {
+  await I.verifyCommand(`docker run -d --restart always -e PERCONA_TEST_PLATFORM_ADDRESS=https://check-dev.percona.com:443 -e PMM_PUBLIC_ADDRESS=${publicAddress} --publish 8085:80 --publish 8443:443 --name pmm-server-public-address percona/pmm-server:latest`);
+  await I.verifyCommand('docker exec pmm-server-public-address yum update -y percona-release');
+  await I.verifyCommand('docker exec pmm-server-public-address sed -i\'\' -e \'s^/release/^/experimental/^\' /etc/yum.repos.d/pmm2-server.repo');
+  await I.verifyCommand('docker exec pmm-server-public-address percona-release enable percona experimental');
+  await I.verifyCommand('docker exec pmm-server-public-address yum clean all');
+  await I.verifyCommand('docker restart pmm-server-public-address');
+  await I.wait(30);
+};
+
+Before(async ({ I, portalAPI }) => {
+  portalUser = await portalAPI.getUser();
+  await portalAPI.oktaCreateUser(portalUser);
+  adminToken = await portalAPI.getUserAccessToken(portalUser.email, portalUser.password);
+  freeOrg = await portalAPI.apiCreateOrg(adminToken);
+
+  await I.Authorize('admin', 'admin');
+});
+
+After(async ({ I, portalAPI }) => {
+  await portalAPI.apiDeleteOrg(freeOrg.id, adminToken);
+  await portalAPI.oktaDeleteUserByEmail(portalUser.email);
+  await I.verifyCommand('docker stop pmm-server-public-address');
+  await I.verifyCommand('docker rm pmm-server-public-address');
+});
+
+Data(publicIPs).Scenario(
+  'PMM-T1173 PMM-T1174 Verify PMM_PUBLIC_ADDRESS env variable with IP @docker-configuration',
+  async ({
+    I, pmmSettingsPage, current, perconaPlatformPage,
+  }) => {
+    const basePmmUrl = 'http://127.0.0.1:8085/';
+    const { publicAddress } = current;
+
+    await runContainerWithPublicAddressVariable(I, publicAddress);
+    await I.amOnPage(basePmmUrl + pmmSettingsPage.advancedSettingsUrl);
+    await I.waitForVisible(pmmSettingsPage.fields.publicAddressInput, 30);
+    const setPublicAddress = await I.grabValueFrom(pmmSettingsPage.fields.publicAddressInput);
+
+    assert.ok(setPublicAddress === publicAddress, 'Set public address does not equal to one specified in public address environment variable');
+
+    await I.amOnPage(basePmmUrl + perconaPlatformPage.url);
+    I.waitForVisible(perconaPlatformPage.elements.settingsContent, 30);
+    await perconaPlatformPage.connectToPortal(adminToken);
+  },
+);
+
+Scenario(
+  'PMM-T1176 Verify PMM_PUBLIC_ADDRESS env variable after upgrade @docker-configuration',
+  async ({
+    I, pmmSettingsPage, homePage,
+  }) => {
+    const basePmmUrl = 'http://127.0.0.1:8085/';
+
+    await runContainerWithPublicAddressVariableUpgrade(I, '127.0.0.1');
```

Comment:
could you not set variable for the public address and use it everywhere in the test?

Reply:
Changed